src/HPolygon.jl

```diff
@@ -29,24 +29,31 @@ struct HPolygon{N<:Real} <: AbstractHPolygon{N}
 
     # default constructor that applies sorting of the given constraints
     function HPolygon{N}(constraints::Vector{LinearConstraint{N}};
-                         sort_constraints::Bool=true) where {N<:Real}
+                         sort_constraints::Bool=true,
+                         validate_boundedness::Bool=false) where {N<:Real}
         if sort_constraints
             sorted_constraints = Vector{LinearConstraint{N}}()
             sizehint!(sorted_constraints, length(constraints))
             for ci in constraints
                 addconstraint!(sorted_constraints, ci)
             end
-            return new{N}(sorted_constraints)
+            P = new{N}(sorted_constraints)
         else
-            return new{N}(constraints)
+            P = new{N}(constraints)
         end
+        @assert (!validate_boundedness ||
+                 LazySets.validate_boundedness(P)) "the polygon is not bounded"
```

Comment:
i think that `LazySets.` is not needed here

Reply:
i don't understand the logic behind

```julia
 @assert (!validate_boundedness ||
                 LazySets.validate_boundedness(P)) "the polygon is not bounded"
```

by default `validate_boundedness = false`, meaning that one doesn't want to check if the input polygon is indeed bounded. but then `!validate_boundedness` evaluates to `true` and `LazySets.validate_boundedness(P)` is called.

Reply:
> i think that LazySets. is not needed here

true

Reply:
> but then `!validate_boundedness` evaluates to `true`

correct

> and `LazySets.validate_boundedness(P)` is called

incorrect: The `||` is already satisfied by `true` above, so short-circuiting will terminate the evaluation.

Reply:
> fewer names sounds better

yes. but since one is a kwarg and another a function, we can have them distinct, and we could use:

- `check_boundedness` in the kwarg
-  `validate(S, isbounded)` for validation

for the function i really like: `isbounded(S, use_type_assumption::Bool=true)`, so no new function name needs to be added and one can control the behavior if `isbounded` in a program with one external flag.

Reply:
> for the function i really like: `isbounded(S, use_type_assumption::Bool=true)`

okay